docs/common/js/search.js

```diff
@@ -0,0 +1,43 @@
+$(document).ready(function(){
+  // docgenerate returning undefined objects when creating search index files. TEMP FIX
+  const steps = window.searchIndex.filter(function(n){ return n != undefined })
+
+  var options = {
+    shouldSort: true,
+    includeMatches: true,
+    includeScore: true,
+    threshold: 0.25,
+    tokenize: true,
+    location: 0,
+    distance: 100000,
+    maxPatternLength: 32,
+    matchAllTokens: true,
+    minMatchCharLength: 3,
+    keys: [{
+      name: 'Title',
+      weight: 0.8
+    },{
+      name: 'Snippet',
+      weight: 0.6
+    },{
+      name: 'Body',
+      weight: 0.6
+    }]
+  }
+  var fuse = new Fuse(steps, options)
+  var query = decodeURIComponent(window.location.search.slice(3).split('+').join(' '))
+  var result = fuse.search(query.removeStopWords())
+
+  $('#search-results').empty()
+  $('#search-results').append('<div id="search-result">Search results for: <b>' + query + '</b></div>')
```

Comment:
We can't use an `id` more than once on any given page. These should be `class="search-result"` instead (and we'll have to update the CSS accordingly).